integrationTests/relayers/slowTests/framework/testSetup.go

```diff
@@ -461,45 +469,54 @@ func (setup *TestSetup) IsTransferDoneFromMultiversX(sender KeysHolder, receiver
 }
 
 func (setup *TestSetup) isTransferDoneFromMultiversXForToken(sender, receiver KeysHolder, params TestTokenParams) bool {
-	setup.mutBalances.Lock()
-	initialBalance := setup.esdtBalanceForSafe[params.AbstractTokenIdentifier]
-	setup.mutBalances.Unlock()
-
 	if !setup.checkHolderMvxBalanceForToken(sender, true, params) {
 		return false
 	}
 	if !setup.checkHolderEthBalanceForToken(receiver, false, params) {
 		return false
 	}
 
+	if !setup.checkMvxBalanceForSafe(params) {
+		return false
+	}
+
+	if !setup.checkTokenOnMvxFirstBridge(params) {
+		return false
+	}
+
+	return setup.checkTokenOnEthSecondBridge(params)
+}
+
+func (setup *TestSetup) checkMvxBalanceForSafe(params TestTokenParams) bool {
+	setup.mutBalances.Lock()
+	initialBalance := setup.esdtBalanceForSafe[params.AbstractTokenIdentifier]
+	setup.mutBalances.Unlock()
+
 	expectedBalance := new(big.Int).Add(initialBalance, params.ESDTSafeExtraBalance)
 	actualBalance := setup.MultiversxHandler.GetESDTChainSpecificTokenBalance(
 		setup.Ctx, setup.MultiversxHandler.SafeAddress, params.AbstractTokenIdentifier,
 	)
-	if expectedBalance.Cmp(actualBalance) != 0 {
-		return false
-	}
 
+	return expectedBalance.Cmp(actualBalance) == 0
+}
+
+func (setup *TestSetup) checkTokenOnMvxFirstBridge(params TestTokenParams) bool {
 	if params.IsMintBurnOnMvX {
-		if !setup.checkMvxBurnedTokenBalance(params) {
-			return false
-		}
-	} else if !setup.checkMvxLockedBalanceForToken(params, false) {
-		return false
+		return setup.checkMvxBurnedTokenBalance(params)
 	}
 
+	return setup.checkMvxLockedBalanceForToken(params, secondBridge)
```

Comment:
mismatch between the function name and this provided parameter. Also on the next function